Arion/include/Arion/ExpandingPolytopeAlgorithm.hpp

```diff
@@ -152,16 +158,17 @@ ContactManifold CalculateContactManifold(
             return a.GetHyperPlane().GetDistance() < b.GetHyperPlane().GetDistance();
         });
         direction = face->GetHyperPlane().GetNormal();
+        assert(!glm::isnan(direction.x));
         closestFaceDistance = face->GetHyperPlane().Distance({0, 0, 0});
 
         //Find next CSO point using new search direction
-        glm::dvec3 const aSupportVertex = cso::Support(aShape,  direction);
-        glm::dvec3 const bSupportVertex = cso::Support(bShape, -direction);
-        glm::dvec3 const supportVertex  = cso::Support(aShape, bShape, direction);
-        double const supportVertexDistance = glm::abs(glm::dot(supportVertex, direction));
+        glm::vec3 const aSupportVertex = cso::Support(aShape,  direction);
+        glm::vec3 const bSupportVertex = cso::Support(bShape, -direction);
+        glm::vec3 const supportVertex  = cso::Support(aShape, bShape, direction);
+        float const supportVertexDistanceSing = glm::dot(supportVertex, direction);
```

Comment:
```suggestion
        float const supportVertexDistanceSigned = glm::dot(supportVertex, direction);
```
Not sure what `Sing` meant, I assumed `Signed`